templates/render_warnings.mustache

```diff
@@ -0,0 +1,57 @@
+{{!
+    This file is part of the QuestionPy Moodle plugin - https://questionpy.org
+ 
+    Moodle is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+ 
+    Moodle is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+ 
+    You should have received a copy of the GNU General Public License
+    along with Moodle.  If not, see <http://www.gnu.org/licenses />.
+!}}
+{{!
+    @template qtype_questionpy/render_error
+
+    Views the a render error.
+
+    Context variables required for this template:
+    * message - the error message
+    * info - more informations about the error
+
+    Example context (json):
+    {
+        "message":"This is an error message.",
+        "info": "Here are more informations about the error."
+    }
+}}
+<div class="alert alert-warning">
+    {{#should_use_list}}
+        <ul>
+            {{#warnings}}
+                <li>
+                    {{{localize}}}
+                    <input type="hidden" name="{{#get_qt_field_name}} {{name}} {{/get_qt_field_name}}"
+                           value="{{value}}">
```

Comment:
Ich finde es komisch, innerhalb der Warnung einen hidden input hinzuzufügen. Zur Zeit mag es nur Warnungen vom Typ `invalid_option_warning` geben, sodass es die Felder `name` und `value` immer gibt, aber wird das so bleiben? Ich denke diese Inputs sollten außerhalb des Templates (zum Beispiel in der `renderer.php` oder direkt im `question_ui_renderer`) hinzugefügt werden.

Außerdem: Bei `select`-Elementen werden die fehlenden Optionen doch bereits durch den `question_ui_renderer` hinzugefügt. Ist das nicht redundant?

Reply:
Ersteres fand ich nicht so schlimm, aber zweiteres ist in der Tat ein Problem. 917ef53ae3703cc09beb5bdd6558119ea1736fb7 löst beides.